From e00e49ef2c5de2187305d37e2a942ab5623aac94 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Wed, 22 Nov 2017 18:40:11 +0800
Subject: [PATCH 0671/1051] arm64: dts: marvell: align Armada8040 default dts
 to setup A DB board

This Armada8040 default dts is aligned to setup A DB board topology.
Besides of the CP0 and CP1 comphy setting, setup A DB board has
following IOs: AP_UART, AP_eMMC, CP0_I2C, CP0_eMMC, CP0_RGMII1,
CP1_SPI(Boot), CP1_RGMII0.

Change-Id: I6a7a623a12171ee5c8b4406bed6a4fa0b627de3c
Signed-off-by: Victor Gu <xigu@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../arm64/boot/dts/marvell/armada-8040-db.dts | 61 ++-----------------
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 11 ++++
 2 files changed, 15 insertions(+), 57 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dts b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
index 20698aa17481..3b42efbf9c72 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dts
@@ -9,9 +9,10 @@
 #include "armada-8040-db.dtsi"
 
 / {
-	model = "Marvell Armada 8040 DB board";
-	compatible = "marvell,armada8040-db", "marvell,armada8040",
-		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+	model = "Marvell Armada 8040 development board default (A) setup";
+	compatible = "marvell,armada8040-db-default", "marvell,armada8040-db",
+		"marvell,armada8040", "marvell,armada-ap806-quad",
+		"marvell,armada-ap806";
 
 	chosen {
 		stdout-path = "serial0:115200n8";
@@ -30,33 +31,6 @@
 	};
 };
 
-&spi0 {
-	status = "okay";
-
-	spi-flash@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <10000000>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			partition@0 {
-				label = "U-Boot";
-				reg = <0 0x200000>;
-			};
-			partition@400000 {
-				label = "Filesystem";
-				reg = <0x200000 0xce0000>;
-			};
-		};
-	};
-};
-
 /* Accessible over the mini-USB CON9 connector on the main board */
 &uart0 {
 	status = "okay";
@@ -173,33 +147,6 @@
 &cp1_nand_controller {
 	pinctrl-0 = <&nand_pins>, <&nand_rb>;
 	pinctrl-names = "default";
-
-	nand@0 {
-		reg = <0>;
-		nand-rb = <0>;
-		nand-on-flash-bbt;
-		nand-ecc-strength = <4>;
-		nand-ecc-step-size = <512>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			partition@0 {
-				label = "U-Boot";
-				reg = <0 0x200000>;
-			};
-			partition@200000 {
-				label = "Linux";
-				reg = <0x200000 0xe00000>;
-			};
-			partition@1000000 {
-				label = "Filesystem";
-				reg = <0x1000000 0x3f000000>;
-			};
-		};
-	};
 };
 
 &cp1_sata0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index ded455ee43f7..b136f0008907 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -315,6 +315,17 @@
 			clocks = <&CP110_LABEL(clk) 1 15>,
 				 <&CP110_LABEL(clk) 1 16>;
 			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			sata-port@0 {
+				reg = <0>;
+				satus = "disabled";
+			};
+			sata-port@1 {
+				reg = <1>;
+				status = "disabled";
+			};
 		};
 
 		CP110_LABEL(xor0): xor@6a0000 {
-- 
2.17.1

